File: app/src/main/java/com/kelsos/mbrc/commands/HandleHandshake.java
package com.kelsos.mbrc.commands;

import javax.inject.Inject;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.model.ConnectionModel;
import com.kelsos.mbrc.services.ProtocolHandler;

public class HandleHandshake implements ICommand {
  private ProtocolHandler handler;
  private ConnectionModel model;

  @Inject public HandleHandshake(ProtocolHandler handler, ConnectionModel model) {
    this.handler = handler;
    this.model = model;
  }

  public void execute(IEvent e) {
    if (!(Boolean) e.getData()) {
      handler.resetHandshake();
      model.setHandShakeDone(false);
    }
  }
}

File: app/src/main/java/com/kelsos/mbrc/commands/HandleHandshake.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.common.state.ConnectionModel
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolHandler
import com.kelsos.mbrc.networking.protocol.ProtocolMessage
import javax.inject.Inject

class HandleHandshake
  @Inject
  constructor(
    private val handler: ProtocolHandler,
    private val model: ConnectionModel,
  ) : ProtocolAction {
    override fun execute(message: ProtocolMessage) {
      if (!(message.data as Boolean)) {
        handler.resetHandshake()
        model.setHandShakeDone(false)
      }
    }
  }
